python_modules/dagster/dagster/_core/definitions/partition_mapping.py

```diff
@@ -193,6 +197,145 @@ def get_downstream_partitions_for_partition_range(
         raise NotImplementedError()
 
 
+@experimental
+class SingleDimensionToMultiPartitionMapping(PartitionMapping):
```

Comment:
From the name, I wasn't able to tell which direction this goes - i.e. whether the upstream or the downstream partitions definition is expected to be multi-. Is there a way to make that more clear, or at least conform to a convention?

FWIW `AllPartitionMapping` and `LastPartitionMapping`, prefer to what partitions in the upstream asset correspond to a partition of the downstream asset.

Reply:
That's a good point. Renaming this to `SingleDimensionDependencyMapping`, hopefully that is clearer.

Tossing out some possible naming ideas:
- SingleDimensionDependencyMapping
- MatchesDownstreamPartitionDimensionMapping
- PartitionDimensionMapping